client-js/app/elements/labrad-plot.ts

```diff
@@ -811,3 +789,66 @@ function prettyNumber(value: number, min: number, max: number, res: number): str
   }
 }
 
+
+/**
+ * Get a color for the given value on a scale with the given min and max
+ * limits (zMin <= z <= zMax).
+ */
+function getColor(z: number, zMin: number, zMax: number) {
+  var index = 128;
+  if (zMin !== zMax) {
+    index = Math.floor(255 * (z - zMin) / (zMax - zMin));
```

Comment:
I see what you mean. I don't like the appending the same color twice trick at all though. I'd be more inclined to round half-up or down rather than using `Math.floor` to fix this problem.